### ISSUE TITLE
Running out of memory during installation

### ISSUE THREAD

**Issue by julienroy13** [google-research/football#49]
Installation fails at building wheel for gfootball. It seems to be due to out of memory error. Is there a way to limit memory / number of threads used during the installation?

I am running on Ubuntu 18.04 and have 16GB of RAM.

Error output:
```
[ 86%] Building CXX object CMakeFiles/gamelib.dir/src/onthepitch/proceduralpitch.cpp.o
  virtual memory exhausted: Cannot allocate memory
  virtual memory exhausted: Cannot allocate memory
  virtual memory exhausted: Cannot allocate memory
  virtual memory exhausted: Cannot allocate memory
  CMakeFiles/menulib.dir/build.make:110: recipe for target 'CMakeFiles/menulib.dir/src/menu/menutask.cpp.o' failed
  make[2]: *** [CMakeFiles/menulib.dir/src/menu/menutask.cpp.o] Error 1
  make[2]: *** Waiting for unfinished jobs....
  CMakeFiles/gamelib.dir/build.make:62: recipe for target 'CMakeFiles/gamelib.dir/src/onthepitch/officials.cpp.o' failed
  make[2]: *** [CMakeFiles/gamelib.dir/src/onthepitch/officials.cpp.o] Error 1
  make[2]: *** Waiting for unfinished jobs....
  CMakeFiles/gamelib.dir/build.make:446: recipe for target 'CMakeFiles/gamelib.dir/src/onthepitch/player/controller/strategies/offtheball/default_off.cpp.o' failed
  make[2]: *** [CMakeFiles/gamelib.dir/src/onthepitch/player/controller/strategies/offtheball/default_off.cpp.o] Error 1
  CMakeFiles/gamelib.dir/build.make:158: recipe for target 'CMakeFiles/gamelib.dir/src/onthepitch/player/humanoid/humanoid.cpp.o' failed
  make[2]: *** [CMakeFiles/gamelib.dir/src/onthepitch/player/humanoid/humanoid.cpp.o] Error 1
  virtual memory exhausted: Cannot allocate memory
  virtual memory exhausted: Cannot allocate memory
  CMakeFiles/datalib.dir/build.make:110: recipe for target 'CMakeFiles/datalib.dir/src/data/teamdata.cpp.o' failed
  make[2]: *** [CMakeFiles/datalib.dir/src/data/teamdata.cpp.o] Error 1
  make[2]: *** Waiting for unfinished jobs....
  CMakeFiles/systemsgraphicslib.dir/build.make:158: recipe for target 'CMakeFiles/systemsgraphicslib.dir/src/systems/graphics/objects/graphics_light.cpp.o' failed
  make[2]: *** [CMakeFiles/systemsgraphicslib.dir/src/systems/graphics/objects/graphics_light.cpp.o] Error 1
  make[2]: *** Waiting for unfinished jobs....
  virtual memory exhausted: Cannot allocate memory
  virtual memory exhausted: Cannot allocate memory
  CMakeFiles/gamelib.dir/build.make:86: recipe for target 'CMakeFiles/gamelib.dir/src/onthepitch/player/humanoid/humanoid_utils.cpp.o' failed
  make[2]: *** [CMakeFiles/gamelib.dir/src/onthepitch/player/humanoid/humanoid_utils.cpp.o] Error 1
  CMakeFiles/gamelib.dir/build.make:350: recipe for target 'CMakeFiles/gamelib.dir/src/onthepitch/player/controller/elizacontroller.cpp.o' failed
  make[2]: *** [CMakeFiles/gamelib.dir/src/onthepitch/player/controller/elizacontroller.cpp.o] Error 1
  virtual memory exhausted: Cannot allocate memory
  CMakeFiles/loaderslib.dir/build.make:86: recipe for target 'CMakeFiles/loaderslib.dir/src/loaders/aseloader.cpp.o' failed
  make[2]: *** [CMakeFiles/loaderslib.dir/src/loaders/aseloader.cpp.o] Error 1
  make[2]: *** Waiting for unfinished jobs....
  virtual memory exhausted: Cannot allocate memory
  CMakeFiles/gamelib.dir/build.make:686: recipe for target 'CMakeFiles/gamelib.dir/src/onthepitch/team.cpp.o' failed
  make[2]: *** [CMakeFiles/gamelib.dir/src/onthepitch/team.cpp.o] Error 1
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/systemsgraphicslib.dir/build.make:206: recipe for target 'CMakeFiles/systemsgraphicslib.dir/src/systems/graphics/graphics_scene.cpp.o' failed
  make[2]: *** [CMakeFiles/systemsgraphicslib.dir/src/systems/graphics/graphics_scene.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/baselib.dir/build.make:326: recipe for target 'CMakeFiles/baselib.dir/src/base/math/bluntmath.cpp.o' failed
  make[2]: *** [CMakeFiles/baselib.dir/src/base/math/bluntmath.cpp.o] Error 4
  make[2]: *** Waiting for unfinished jobs....
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/systemsgraphicslib.dir/build.make:110: recipe for target 'CMakeFiles/systemsgraphicslib.dir/src/systems/graphics/objects/graphics_geometry.cpp.o' failed
  make[2]: *** [CMakeFiles/systemsgraphicslib.dir/src/systems/graphics/objects/graphics_geometry.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/gamelib.dir/build.make:230: recipe for target 'CMakeFiles/gamelib.dir/src/onthepitch/player/playerbase.cpp.o' failed
  make[2]: *** [CMakeFiles/gamelib.dir/src/onthepitch/player/playerbase.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/scenelib.dir/build.make:374: recipe for target 'CMakeFiles/scenelib.dir/src/scene/resources/geometrydata.cpp.o' failed
  make[2]: *** [CMakeFiles/scenelib.dir/src/scene/resources/geometrydata.cpp.o] Error 4
  make[2]: *** Waiting for unfinished jobs....
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/menulib.dir/build.make:134: recipe for target 'CMakeFiles/menulib.dir/src/menu/menuscene.cpp.o' failed
  make[2]: *** [CMakeFiles/menulib.dir/src/menu/menuscene.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/gamelib.dir/build.make:206: recipe for target 'CMakeFiles/gamelib.dir/src/onthepitch/player/player.cpp.o' failed
  make[2]: *** [CMakeFiles/gamelib.dir/src/onthepitch/player/player.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/menulib.dir/build.make:62: recipe for target 'CMakeFiles/menulib.dir/src/menu/startmatch/loadingmatch.cpp.o' failed
  make[2]: *** [CMakeFiles/menulib.dir/src/menu/startmatch/loadingmatch.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/gamelib.dir/build.make:182: recipe for target 'CMakeFiles/gamelib.dir/src/onthepitch/player/playerofficial.cpp.o' failed
  make[2]: *** [CMakeFiles/gamelib.dir/src/onthepitch/player/playerofficial.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/gamelib.dir/build.make:494: recipe for target 'CMakeFiles/gamelib.dir/src/onthepitch/player/controller/strategies/offtheball/goalie_default.cpp.o' failed
  make[2]: *** [CMakeFiles/gamelib.dir/src/onthepitch/player/controller/strategies/offtheball/goalie_default.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/gui2lib.dir/build.make:86: recipe for target 'CMakeFiles/gui2lib.dir/src/utils/gui2/widgets/caption.cpp.o' failed
  make[2]: *** [CMakeFiles/gui2lib.dir/src/utils/gui2/widgets/caption.cpp.o] Error 4
  make[2]: *** Waiting for unfinished jobs....
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/systemsgraphicslib.dir/build.make:86: recipe for target 'CMakeFiles/systemsgraphicslib.dir/src/systems/graphics/graphics_task.cpp.o' failed
  make[2]: *** [CMakeFiles/systemsgraphicslib.dir/src/systems/graphics/graphics_task.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/menulib.dir/build.make:86: recipe for target 'CMakeFiles/menulib.dir/src/menu/pagefactory.cpp.o' failed
  make[2]: *** [CMakeFiles/menulib.dir/src/menu/pagefactory.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/gamelib.dir/build.make:134: recipe for target 'CMakeFiles/gamelib.dir/src/onthepitch/player/humanoid/humanoidbase.cpp.o' failed
  make[2]: *** [CMakeFiles/gamelib.dir/src/onthepitch/player/humanoid/humanoidbase.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/utilslib.dir/build.make:134: recipe for target 'CMakeFiles/utilslib.dir/src/utils/objectloader.cpp.o' failed
  make[2]: *** [CMakeFiles/utilslib.dir/src/utils/objectloader.cpp.o] Error 4
  make[2]: *** Waiting for unfinished jobs....
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/gui2lib.dir/build.make:254: recipe for target 'CMakeFiles/gui2lib.dir/src/utils/gui2/guitask.cpp.o' failed
  make[2]: *** [CMakeFiles/gui2lib.dir/src/utils/gui2/guitask.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/frameworklib.dir/build.make:86: recipe for target 'CMakeFiles/frameworklib.dir/src/framework/scheduler.cpp.o' failed
  make[2]: *** [CMakeFiles/frameworklib.dir/src/framework/scheduler.cpp.o] Error 4
  make[2]: *** Waiting for unfinished jobs....
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/gui2lib.dir/build.make:230: recipe for target 'CMakeFiles/gui2lib.dir/src/utils/gui2/windowmanager.cpp.o' failed
  make[2]: *** [CMakeFiles/gui2lib.dir/src/utils/gui2/windowmanager.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/baselib.dir/build.make:134: recipe for target 'CMakeFiles/baselib.dir/src/base/log.cpp.o' failed
  make[2]: *** [CMakeFiles/baselib.dir/src/base/log.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/menulib.dir/build.make:182: recipe for target 'CMakeFiles/menulib.dir/src/menu/ingame/gamepage.cpp.o' failed
  make[2]: *** [CMakeFiles/menulib.dir/src/menu/ingame/gamepage.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/gamelib.dir/build.make:110: recipe for target 'CMakeFiles/gamelib.dir/src/onthepitch/player/humanoid/animcollection.cpp.o' failed
  make[2]: *** [CMakeFiles/gamelib.dir/src/onthepitch/player/humanoid/animcollection.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/gamelib.dir/build.make:542: recipe for target 'CMakeFiles/gamelib.dir/src/onthepitch/ball.cpp.o' failed
  make[2]: *** [CMakeFiles/gamelib.dir/src/onthepitch/ball.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/gamelib.dir/build.make:638: recipe for target 'CMakeFiles/gamelib.dir/src/onthepitch/AIsupport/AIfunctions.cpp.o' failed
  make[2]: *** [CMakeFiles/gamelib.dir/src/onthepitch/AIsupport/AIfunctions.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/gamelib.dir/build.make:566: recipe for target 'CMakeFiles/gamelib.dir/src/onthepitch/match.cpp.o' failed
  make[2]: *** [CMakeFiles/gamelib.dir/src/onthepitch/match.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/gamelib.dir/build.make:326: recipe for target 'CMakeFiles/gamelib.dir/src/onthepitch/player/controller/refereecontroller.cpp.o' failed
  make[2]: *** [CMakeFiles/gamelib.dir/src/onthepitch/player/controller/refereecontroller.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/managerslib.dir/build.make:62: recipe for target 'CMakeFiles/managerslib.dir/src/managers/scenemanager.cpp.o' failed
  make[2]: *** [CMakeFiles/managerslib.dir/src/managers/scenemanager.cpp.o] Error 4
  make[2]: *** Waiting for unfinished jobs....
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/scenelib.dir/build.make:302: recipe for target 'CMakeFiles/scenelib.dir/src/scene/scene3d/node.cpp.o' failed
  make[2]: *** [CMakeFiles/scenelib.dir/src/scene/scene3d/node.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/scenelib.dir/build.make:182: recipe for target 'CMakeFiles/scenelib.dir/src/scene/objects/geometry.cpp.o' failed
  make[2]: *** [CMakeFiles/scenelib.dir/src/scene/objects/geometry.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/gamelib.dir/build.make:710: recipe for target 'CMakeFiles/gamelib.dir/src/onthepitch/teamAIcontroller.cpp.o' failed
  make[2]: *** [CMakeFiles/gamelib.dir/src/onthepitch/teamAIcontroller.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/gui2lib.dir/build.make:110: recipe for target 'CMakeFiles/gui2lib.dir/src/utils/gui2/widgets/image.cpp.o' failed
  make[2]: *** [CMakeFiles/gui2lib.dir/src/utils/gui2/widgets/image.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/datalib.dir/build.make:86: recipe for target 'CMakeFiles/datalib.dir/src/data/playerdata.cpp.o' failed
  make[2]: *** [CMakeFiles/datalib.dir/src/data/playerdata.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/scenelib.dir/build.make:230: recipe for target 'CMakeFiles/scenelib.dir/src/scene/objects/joint.cpp.o' failed
  make[2]: *** [CMakeFiles/scenelib.dir/src/scene/objects/joint.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/baselib.dir/build.make:86: recipe for target 'CMakeFiles/baselib.dir/src/base/utils.cpp.o' failed
  make[2]: *** [CMakeFiles/baselib.dir/src/base/utils.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/systemsgraphicslib.dir/build.make:182: recipe for target 'CMakeFiles/systemsgraphicslib.dir/src/systems/graphics/objects/graphics_overlay2d.cpp.o' failed
  make[2]: *** [CMakeFiles/systemsgraphicslib.dir/src/systems/graphics/objects/graphics_overlay2d.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/baselib.dir/build.make:62: recipe for target 'CMakeFiles/baselib.dir/src/base/sdl_surface.cpp.o' failed
  make[2]: *** [CMakeFiles/baselib.dir/src/base/sdl_surface.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/menulib.dir/build.make:158: recipe for target 'CMakeFiles/menulib.dir/src/menu/ingame/radar.cpp.o' failed
  make[2]: *** [CMakeFiles/menulib.dir/src/menu/ingame/radar.cpp.o] Error 4
  c++: internal compiler error: Killed (program cc1plus)
  Please submit a full bug report,
  with preprocessed source if appropriate.
  See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
  CMakeFiles/gamelib.dir/build.make:278: recipe for target 'CMakeFiles/gamelib.dir/src/onthepitch/player/controller/humancontroller.cpp.o' failed
  make[2]: *** [CMakeFiles/gamelib.dir/src/onthepitch/player/controller/humancontroller.cpp.o] Error 4
  [ 86%] Built target systemscommonlib
  [ 86%] Built target typeslib
  CMakeFiles/Makefile2:375: recipe for target 'CMakeFiles/frameworklib.dir/all' failed
  make[1]: *** [CMakeFiles/frameworklib.dir/all] Error 2
  make[1]: *** Waiting for unfinished jobs....
  CMakeFiles/Makefile2:523: recipe for target 'CMakeFiles/loaderslib.dir/all' failed
  make[1]: *** [CMakeFiles/loaderslib.dir/all] Error 2
  CMakeFiles/Makefile2:104: recipe for target 'CMakeFiles/baselib.dir/all' failed
  make[1]: *** [CMakeFiles/baselib.dir/all] Error 2
  CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/scenelib.dir/all' failed
  make[1]: *** [CMakeFiles/scenelib.dir/all] Error 2
  CMakeFiles/Makefile2:178: recipe for target 'CMakeFiles/menulib.dir/all' failed
  make[1]: *** [CMakeFiles/menulib.dir/all] Error 2
  CMakeFiles/Makefile2:141: recipe for target 'CMakeFiles/datalib.dir/all' failed
  make[1]: *** [CMakeFiles/datalib.dir/all] Error 2
  CMakeFiles/Makefile2:449: recipe for target 'CMakeFiles/managerslib.dir/all' failed
  make[1]: *** [CMakeFiles/managerslib.dir/all] Error 2
  CMakeFiles/Makefile2:486: recipe for target 'CMakeFiles/gui2lib.dir/all' failed
  make[1]: *** [CMakeFiles/gui2lib.dir/all] Error 2
  CMakeFiles/Makefile2:412: recipe for target 'CMakeFiles/utilslib.dir/all' failed
  make[1]: *** [CMakeFiles/utilslib.dir/all] Error 2
  CMakeFiles/Makefile2:338: recipe for target 'CMakeFiles/gamelib.dir/all' failed
  make[1]: *** [CMakeFiles/gamelib.dir/all] Error 2
  CMakeFiles/Makefile2:215: recipe for target 'CMakeFiles/systemsgraphicslib.dir/all' failed
  make[1]: *** [CMakeFiles/systemsgraphicslib.dir/all] Error 2
  Makefile:83: recipe for target 'all' failed
  make: *** [all] Error 2
  gfootball/build_game_engine.sh: line 19: pushd: third_party/gfootball_engine: No such file or directory
  Google Research Football compilation failed
  
  ----------------------------------------
  Failed building wheel for gfootball
```

**Comment by Bravlin** [comment]
In gfootball/build_game_engine.sh look for the line where the command make appears. That command is receiving -j as an argument. That's telling it to generate as many threads for compilation as it wants, resulting in the out of memory problem. Just add a number after -j to limit the amount of threads generated to that specific number.

**Comment by julienroy13** [comment]
It solved that issue. Thanks!

**Comment by jorgeg96** [comment]
I think I'm having the same issue. Could you explain in detail how you solved it by limiting the number of threads? Thanks!

**Comment by Bravlin** [comment]
Of course! One of the things the command pip install does in this case is building the gfootball environment. For doing that, it has to call the script build_game_engine.sh, which found in the folder gfootball from the root of the project.

On the line number 18 of that same file, you will find make -j between &&. (&& make -j &&). That line is executing various commands, but we're interested in make in particular. That -j is telling make to build gfootball with parallelism, so it creates various threads to handle the task. The thing is, if a number is not specified after -j then the command will create as many threads as it pleases, quickly consuming the available memory. So, to fix the issue, just add a reasonable number after -j.

**Comment by qstanczyk** [comment]
Fix was just pushed in v1.2 release.
